src/bin/pgcopydb/dump_restore.c

```diff
@@ -711,7 +711,30 @@ copydb_write_restore_list(CopyDataSpec *specs, PostgresDumpSection section)
 			name = NULL;
 		}
 
-		if (!skip && copydb_objectid_is_filtered_out(specs, oid, name))
+		/*
+		 * There could be a case where the materalized view is included in the
+		 * dump, but the refresh is filtered out using [exclude-table-data].
+		 * In this case, we want to skip the refresh.
+		 */
+		if (!skip &&
+			(item->desc == ARCHIVE_TAG_REFRESH_MATERIALIZED_VIEW ||
+			 item->desc == ARCHIVE_TAG_MATERIALIZED_VIEW))
+		{
+			bool isRefresh = item->desc == ARCHIVE_TAG_REFRESH_MATERIALIZED_VIEW;
+
+			if (copydb_matview_objectid_is_filtered_out(specs, oid, isRefresh))
```

Comment:
I was expecting a special processing for `[exclude-table-data]` filtering when the tag is about REFRESH, but the rest of the processing to just be as usual, covered in the filter table and the `copydb_objectid_is_filtered_out()` function call.

Reply:
The reason for having special handling for ARCHIVE_TAG_MATERIALIZED_VIEW is that we also need to exclude ARCHIVE_TAG_REFRESH_MATERIALIZED_VIEW when the materialized view itself is filtered out.

Reply:
Yes that's what I meant. In that case when the matview itself is filtered out, certainly the existing function and filter table are covering all we need, right?

Reply:
The existing function `copydb_objectid_is_filtered_out` filters based on oid and never it considers the fact whether it is a matview itself or just matview refresh.

Reply:
Exactly. So first do a special check for REFRESH, and then let the normal code behave as usual for the CREATE MATERIALIZED VIEW, skipping that only when the OID is found in our filter table. Which means when skipping only the REFRESH parts, we do NOT add the OID of the matview to the filter table.

It seems to me that would be easier to understand and maintain.

Reply:
@dimitri IIUC, you are suggesting not to insert the OID of the materialized view into the filter table if it's data is excluded. Instead, refer to s_matview table to filter the materialized view refresh?

Reply:
Indeed. Keep the current behavior, just add CREATE MAT VIEWS to the filtering when needed, and only add a specific new behavior for REFRESH, which is exclude-data, and new, and needs special care.